extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleJobStore.java

```diff
@@ -325,21 +302,22 @@ private Key getDataKey(UUID jobId, String key) {
   }
 
   @Override
-  public <T extends DataModel> T findData(UUID jobId, String key, Class<T> type) {
-    Key entityKey = getDataKey(jobId, key);
-    Entity entity = datastore.get(entityKey);
-    if (entity == null) {
-      return null;
-    }
-    try {
-      return objectMapper.readValue(entity.getString(type.getName()), type);
-    } catch (IOException t) {
-      throw new RuntimeException("Failed to deserialized entityKey: " + entityKey, t);
-    }
+  public <T extends DataModel> T findData(UUID jobId, String key, Class<T> type)
+      throws IOException {
+    return getItem(jobId, key, type);
   }
 
   @Override
   public void create(UUID jobId, String key, InputStream stream) throws IOException {
+    storeNewItem(jobId, key, stream);
+  }
+
+  @Override
+  public InputStream getStream(UUID jobId, String key) throws IOException {
+    return getItem(jobId, key, InputStream.class);
+  }
+
+  private void storeNewItem(UUID jobId, String key, Object object) throws IOException {
```

Comment:
I think generalizing this input parameter to type Object is error prone since anything can be passed in.

How about instead we pass in a serialized model (type String), which is the result of ObjectMapper.writeAs...?

Reply:
Done.